polymetis/polymetis/python/polysim/envs/habitat_manipulator.py

```diff
@@ -0,0 +1,286 @@
+    """
+    if not os.path.isabs(glb_path):
+        glb_path = os.path.join(habitat_dir, glb_path)
+
+    # simulator configuration
+    backend_cfg = habitat_sim.SimulatorConfiguration()
+    backend_cfg.scene_id = glb_path
+    backend_cfg.enable_physics = True
+    backend_cfg.physics_config_file = os.path.join(
+        habitat_dir, backend_cfg.physics_config_file
+    )
+
+    # sensor configurations
+    # Note: all sensors must have the same resolution
+    # setup 2 rgb sensors for 1st and 3rd person views
+    camera_resolution = [540, 720]
+    sensors = {
+        "rgba_camera_1stperson": {
+            "sensor_type": habitat_sim.SensorType.COLOR,
+            "resolution": camera_resolution,
+            "position": [0.0, 0.0, 0.0],
+            "orientation": [0.0, 0.0, 0.0],
+        },
+        "depth_camera_1stperson": {
+            "sensor_type": habitat_sim.SensorType.DEPTH,
+            "resolution": camera_resolution,
+            "position": [0.0, 0.0, 0.0],
+            "orientation": [0.0, 0.0, 0.0],
+        },
+    }
+
+    sensor_specs = []
+    for sensor_uuid, sensor_params in sensors.items():
+        sensor_spec = habitat_sim.CameraSensorSpec()
+        sensor_spec.uuid = sensor_uuid
+        sensor_spec.sensor_type = sensor_params["sensor_type"]
+        sensor_spec.resolution = sensor_params["resolution"]
+        sensor_spec.position = sensor_params["position"]
+        sensor_spec.orientation = sensor_params["orientation"]
+        sensor_specs.append(sensor_spec)
+
+    # agent configuration
+    agent_cfg = habitat_sim.agent.AgentConfiguration()
+    agent_cfg.sensor_specifications = sensor_specs
+
+    return habitat_sim.Configuration(backend_cfg, [agent_cfg])
+
+
+def place_agent(sim: habitat_sim.Simulator) -> magnum.Matrix4:
+    """Sets AgentState to some reasonable values and return a transformation matrix."""
+    # place our agent in the scene
+    agent_state = habitat_sim.AgentState()
+    agent_state.position = [-0.15, -0.1, 1.0]
+    # agent_state.position = [-0.15, -1.6, 1.0]
+    agent_state.rotation = np.quaternion(-0.83147, 0, 0.55557, 0)
+    agent = sim.initialize_agent(0, agent_state)
+    return agent.scene_node.transformation_matrix()
+
+
+def place_robot_from_agent(
+    sim: habitat_sim.Simulator,
+    robot: habitat_sim._ext.habitat_sim_bindings.ManagedBulletArticulatedObject,
+    angle_correction: float = -1.56,
+    local_base_pos: np.ndarray = None,
+) -> None:
+    """Moves robot to reasonable transformation relative to agent."""
+    if local_base_pos is None:
+        local_base_pos = np.array([0.0, -1.1, -2.0])
+    # place the robot root state relative to the agent
+    agent_transform = sim.agents[0].scene_node.transformation_matrix()
+    base_transform = mn.Matrix4.rotation(
+        mn.Rad(angle_correction), mn.Vector3(1.0, 0, 0)
+    )
+    base_transform.translation = agent_transform.transform_point(local_base_pos)
+    robot.transformation = base_transform
```

Comment:
Are the values for agent state and base position specific to the Franka?
If so maybe they should be set from the configs?

Reply:
While not specific to the Franka, they are specific to the scene. I put them as optional parameters in HabitatManipulator but did not add them to the yaml configs, since this should probably be tweaked in detail when someone iterates on specific Habitat scenes.